code/modules/clothing/head/perceptomatrix.dm

```diff
@@ -0,0 +1,202 @@
+		/datum/component/anti_magic, \
+		antimagic_flags = MAGIC_RESISTANCE_MIND, \
+		inventory_flags = ITEM_SLOT_HEAD, \
+	))
+
+	update_icon_state()
+
+/obj/item/clothing/head/helmet/perceptomatrix/Destroy(force)
+	QDEL_NULL(transform_action)
+	QDEL_NULL(active_components)
+	return ..()
+
+/obj/item/clothing/head/helmet/perceptomatrix/examine(mob/user)
+	. = ..()
+	if (!core_installed)
+		. += span_warning("It requires a hallucination anomaly core in order to function.")
+
+/obj/item/clothing/head/helmet/perceptomatrix/item_action_slot_check(slot, mob/user, datum/action/action)
+	return slot & ITEM_SLOT_HEAD
+
+/obj/item/clothing/head/helmet/perceptomatrix/update_icon_state()
+	icon_state = base_icon_state + (core_installed ? "" : "_inactive")
+	worn_icon_state = base_icon_state + (core_installed ? "" : "_inactive")
+	return ..()
+
+/obj/item/clothing/head/helmet/perceptomatrix/attackby(obj/item/weapon, mob/user, params)
+	if (!istype(weapon, /obj/item/assembly/signaler/anomaly/hallucination))
+		return ..()
+	balloon_alert(user, "inserting...")
+	if (!do_after(user, delay = 3 SECONDS, target = src))
+		return
+	qdel(weapon)
+	core_installed = TRUE
+	update_anomaly_state()
+	update_appearance(UPDATE_ICON_STATE)
+	playsound(src, 'sound/machines/crate/crate_open.ogg', 50, FALSE)
+
+/// Pre-activated polymorph belt
+/obj/item/clothing/head/helmet/perceptomatrix/functioning
+	core_installed = TRUE
+
+/datum/action/cooldown/spell/pointed/percept_hallucination
+	name = "Hallucinate"
+	desc = "Redirect perceptual energies towards a target, staggering them."
+	button_icon_state = "blind"
+	ranged_mousepointer = 'icons/effects/mouse_pointers/blind_target.dmi'
+
+	sound = 'sound/items/weapons/emitter2.ogg'
+	school = SCHOOL_PSYCHIC
+	cooldown_time = 25 SECONDS
+
+	invocation_type = INVOCATION_NONE
+	spell_requirements = NONE
+	antimagic_flags = MAGIC_RESISTANCE_MIND
+
+	active_msg = "You prepare to zap a target with hallucinations..."
+
+	/// The amount of blurriness to apply
+	var/eye_blur_duration = 7 SECONDS
+	/// The amount of stagger to apply
+	var/stagger_duration = 3 SECONDS
+	/// The amount of hallucination to apply
+	var/hallucination_duration = 25 SECONDS
+	/// Helmet the spell's bound to.
+	var/obj/item/clothing/head/helmet/perceptomatrix/linked_helmet
+	var/datum/effect_system/spark_spread/quantum/spark_sys = new /datum/effect_system/spark_spread/quantum
+
+/datum/action/cooldown/spell/pointed/percept_hallucination/New(Target)
+	. = ..()
+
+	linked_helmet = Target
+
+	spark_sys = new /datum/effect_system/spark_spread/quantum
+
+
+/datum/action/cooldown/spell/pointed/percept_hallucination/is_valid_target(atom/cast_on)
+	. = ..()
+	if(!.)
+		return FALSE
+	if(!ishuman(cast_on))
+		return FALSE
+
+	return cast_on
+
+/datum/action/cooldown/spell/pointed/percept_hallucination/can_cast_spell(feedback = TRUE)
+	. = ..()
+	if(!.)
+		return .
+
+	if(!linked_helmet)
+		stack_trace("casting w/o linked perceptomatrix!")
```

Comment:
It seems pretty easy to support this spell being usable without a helmet for adminbus:
```dm
if(!istype(target, /obj/item/clothing/head/helmet/perceptomatrix))
    return TRUE

var/obj/item/clothing/head/helmet/perceptomatrix/helmet = target
if(!helmet.core_installed)
    if(feedback)
        to_chat(owner, span_warning("You must install a hallucination core first!"))
    return  FALSE

return TRUE
```

Reply:
In fact if you want to be DCS pilled, you could have the helmet register a signal on the spell to block the cast if no core is installed

That way you have 0 snowflake code in the spell itself (it would just exist as an independent, usable spell, which is neat)